task-NXDOC-2663-Document-the-nuxeo-keycloak-package (#2237)
nuxeo/doc.nuxeo.com-content: src/nxdoc/nuxeo-add-ons/nuxeo-keycloak.md

---
title: Nuxeo Keycloak
description: Nuxeo Keycloak allows you to integrate Keycloak, an Open Source Identity and Access Management solution.
---

> 

Reviewer: Maybe add a shot paragraph explaining the purpose of the addon, like for the other ones (e.g. Kerberos).

{{#> callout type='info'}}

> See the [Nuxeo Keycloak Package README.md](https://github.com/nuxeo/nuxeo-lts/blob/2023/packages/nuxeo-keycloak-package/README.md) for a complete list of required and optional configs directly contributable via nuxeo.conf

Reviewer: ```suggestion
See the [Nuxeo Keycloak Package README.md](https://github.com/nuxeo/nuxeo-lts/blob/2023/packages/nuxeo-keycloak-package/README.md) for a complete list of required and optional configuration parameters directly contributable via nuxeo.conf
```

{{/callout}}

> ## Nuxeo/Keycloak automation and rest api status code specs

Reviewer: ```suggestion
## Nuxeo/Keycloak Automation and REST API Status Code Specs
```

> Here are the status codes to expect depending on the plugins order in the automation and RestAPI specific auth chains.

Reviewer: ```suggestion
Here are the status codes to expect depending on the plugins order in the `Automation` and `RestAPI` specific authentication chains.
```

<div class="table-scroll">
  <table class="hover">
    <thead>
      <tr>

>         <th colspan="1">chain order</th>

Reviewer: ```suggestion
        <th colspan="1">Chain order</th>
```

        <th colspan="1">Keycloak basic auth enabled</th>

>         <th colspan="1">provided credentials</th>

Reviewer: ```suggestion
        <th colspan="1">Provided credentials</th>
```

      <tr>
        <td>KEYCLOAK_AUTH AUTOMATION_BASIC_AUTH</td>
        <td>true</td>
        <td>none</td>
        <td>302 - Redirect</td>
      </tr>
      <tr>
        <td>KEYCLOAK_AUTH AUTOMATION_BASIC_AUTH</td>
        <td>true</td>
        <td>KO (nuxeo basic)</td>
        <td>401 - Unauthorized</td>
      </tr>
      <tr>
        <td>KEYCLOAK_AUTH AUTOMATION_BASIC_AUTH</td>
        <td>true</td>
        <td>OK (nuxeo basic)</td>
        <td>200 - OK</td>
      </tr>
      <tr>
        <td>KEYCLOAK_AUTH AUTOMATION_BASIC_AUTH</td>
        <td>true</td>
        <td>KO (Keycloak basic)</td>
        <td>401 - Unauthorized</td>
      </tr>
      <tr>
        <td>KEYCLOAK_AUTH AUTOMATION_BASIC_AUTH</td>
        <td>true</td>
        <td>OK (Keycloak basic)</td>
        <td>200 - OK</td>
      </tr>
      <tr>
        <td>KEYCLOAK_AUTH AUTOMATION_BASIC_AUTH</td>
        <td>true</td>
        <td>KO (Keycloak bearer)</td>
        <td>401 - Unauthorized</td>
      </tr>
      <tr>
        <td>KEYCLOAK_AUTH AUTOMATION_BASIC_AUTH</td>
        <td>true</td>
        <td>OK (Keycloak bearer)</td>
        <td>200 - OK</td>
      </tr>
    </tbody>
  </table>
</div>

> ## Sample contribution to change the auth plugins order, overriding the Keycloak package's contribution

Reviewer: Capitalize

Author: I never noticed we did that!

> Here is a sample contribution allowing to put Keycloak first in the automation and RestAPI specific chains.

Reviewer: ```suggestion
Here is a sample contribution allowing to put Keycloak first in the Automation and RestAPI specific chains.
```

---
title: Nuxeo Keycloak
description: Nuxeo Keycloak allows you to integrate Keycloak, an Open Source Identity and Access Management solution.
---

{{! excerpt}}
The [Nuxeo Keycloak package](https://connect.nuxeo.com/nuxeo/site/marketplace/package/nuxeo-keycloak) allows you to integrate [Keycloak](https://www.keycloak.org/), an Open Source Identity and Access Management solution.
{{! /excerpt}}

{{#> callout type='info'}}

> See the [Nuxeo Keycloak Package README.md](https://github.com/nuxeo/nuxeo-lts/blob/2023/packages/nuxeo-keycloak-package/README.md) for a complete list of required and optional configuration parameters directly contributable via nuxeo.conf

Reviewer 2: ```suggestion
See the [Nuxeo Keycloak Package README.md](https://github.com/nuxeo/nuxeo-lts/blob/2023/packages/nuxeo-keycloak-package/README.md) for a complete list of required and optional configuration parameters directly contributable via `nuxeo.conf`.
```
